checkov/terraform/tf_parser.py

```diff
@@ -384,17 +381,17 @@ def get_file_key_with_nested_data(self, file: str, nested_data: Optional[dict[st
             return file
         nested_str = self.get_file_key_with_nested_data(nested_data.get("file"), nested_data.get('nested_modules_data'))
         nested_module_index = nested_data.get('module_index')
-        return get_tf_definition_key_from_module_dependency(file, nested_str, nested_module_index)
+        return get_tf_definition_object_from_module_dependency(file, nested_str, nested_module_index)
```

Comment:
instead of this recursion, you can run over the `nested_data` and add this to the key.tf_source_modules list